test/AWS.Deploy.CLI.UnitTests/SetOptionSettingTests.cs

```diff
@@ -65,7 +65,31 @@ public void SetOptionSettingTests_AllowedValues()
         [Fact]
         public void SetOptionSettingTests_MappedValues()
         {
-            Assert.Throws<InvalidOverrideValueException>(() => _optionSetting.SetValueOverride(_optionSetting.ValueMapping.Values.First()));
+            var recommendation = _recommendations.First(r => r.Recipe.Id == Constants.ASPNET_CORE_BEANSTALK_RECIPE_ID);
+
+            var optionSetting = recommendation.Recipe.OptionSettings.First(x => x.Id.Equals("EnvironmentType"));
+            Assert.Throws<InvalidOverrideValueException>(() => optionSetting.SetValueOverride(optionSetting.ValueMapping.Values.First()));
+        }
+
+        [Fact]
+        public void SetOptionSettingTests_KeyValueType()
+        {
+            var recommendation = _recommendations.First(r => r.Recipe.Id == Constants.ASPNET_CORE_BEANSTALK_RECIPE_ID);
+
+            var optionSetting = recommendation.Recipe.OptionSettings.First(x => x.Id.Equals("ElasticBeanstalkEnvironmentVariables"));
+            var values = new Dictionary<string, string>() { { "key", "value" } };
+            optionSetting.SetValueOverride(values);
```

Comment:
Does this exercise the change in src/AWS.Deploy.Common/Recommendation.cs on line 120, or does SetValueOverride need to be called with a valid json string in order to fire that code? I have limited understanding of this repo -- I'm guessing passing in valid json would simulate how the Toolkit is using this cli.

Reply:
It does exercise the change. The ID we are passing in is of KeyValue type which will trigger the section in SetValueOverride

Reply:
it looks like this code (above the change of interest) would be exercised (due to `valueOverride is Dictionary<string, string>`) :
```
            if (valueOverride is bool || valueOverride is int || valueOverride is long || valueOverride is double || valueOverride is Dictionary<string, string>)
            {
                _valueOverride = valueOverride;
            }
```
What am I missing?

Reply:
when it is being sent by server mode, it is not being recognized as a Dictionary. It is being detected as a string.

Reply:
That brings us to my initial question: can we make a test that calls SetValueOverride with a string (where the string is equivalent to the dictionary), so that we simulate the server mode case? Or is there something about the system under test that is being created in console mode?

Reply:
I added a test case that passes in a serialized dictionary.